ipie/estimators/energy.py

```diff
@@ -157,3 +157,35 @@ def post_reduce_hook(self, data):
         data[ix_nume] = data[ix_nume] / data[ix_deno]
         ix_nume = self._data_index["E2Body"]
         data[ix_nume] = data[ix_nume] / data[ix_deno]
+
+
+class EnergyEstimatorFP(EnergyEstimator):
+    def __init__(
+        self,
+        system=None,
+        ham=None,
+        trial=None,
+        filename=None,
+    ):
+        super().__init__(system, ham, trial, filename)
+
+    def compute_estimator(self, system, walkers, hamiltonian, trial, istep=1):
+        trial.calc_greens_function(walkers)
+        # Need to be able to dispatch here
+        energy = local_energy(system, hamiltonian, walkers, trial)
+        self._data["ENumer"] = xp.sum(walkers.weight * walkers.phase * walkers.ovlp * energy[:, 0])
+        self._data["EDenom"] = xp.sum(walkers.weight * walkers.phase * walkers.ovlp)
+        self._data["E1Body"] = xp.sum(walkers.weight * walkers.phase * walkers.ovlp * energy[:, 1])
+        self._data["E2Body"] = xp.sum(walkers.weight * walkers.phase * walkers.ovlp * energy[:, 2])
+
+        return self.data
+
+    def post_reduce_hook(self, data):
```

Comment:
Isn't this the same function as what is in EnergyEstimator (you are inheriting from it)?
Do we need to define this again?

Reply:
got rid of this